File: src/main/java/javabot/commands/OperationsCommand.java
package javabot.commands;

import java.util.List;

import javabot.Javabot;
import javabot.BotEvent;
import javabot.Message;

/**
 * Created Jan 26, 2009
 *
 * @author <a href="mailto:jlee@antwerkz.com">Justin Lee</a>
 */
public abstract class OperationsCommand implements Command {
    protected void listCurrent(final Javabot bot, final BotEvent event) {
        bot.postMessage(new Message(event.getChannel(), event, "I am currently running the following operations:"));
        bot.postMessage(new Message(event.getChannel(), event, stringify(bot.listActiveOperations())));
    }

    protected String stringify(final List<String> list) {
        final StringBuilder builder = new StringBuilder();
        for (final String operation : list) {
            if(builder.length() != 0) {
                builder.append(", ");
            }
            builder.append(operation);
        }

        return builder.toString();
    }
}

File: src/main/kotlin/javabot/commands/OperationsCommand.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import jakarta.inject.Inject
import java.lang.String.join
import java.util.ArrayList
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao

abstract class OperationsCommand @Inject constructor(bot: Javabot, adminDao: AdminDao) :
    AdminCommand(bot, adminDao) {
    protected fun listCurrent(event: Message, responses: ArrayList<Message>) {
        responses.add(Message(event, Sofia.adminRunningOperations(event.user.nick)))
        responses.add(Message(event, join(",", bot.activeOperations.map({ it.getName() }))))
    }
}
